src/Stratis.Features.FederatedPeg/Collateral/CollateralChecker.cs

```diff
@@ -86,62 +86,48 @@ public async Task InitializeAsync()
                 .Cast<CollateralFederationMember>().Where(x => x.CollateralAmount != null && x.CollateralAmount > 0))
             {
                 this.logger.LogDebug("Initializing federation member {0} with amount {1}.", federationMember.CollateralMainchainAddress, federationMember.CollateralAmount);
-
                 this.depositsByAddress.Add(federationMember.CollateralMainchainAddress, 0);
             }
 
-            while (true)
+            while (!this.cancellationSource.IsCancellationRequested && !this.collateralUpdated)
             {
-                bool success = await this.UpdateCollateralInfoAsync(this.cancellationSource.Token).ConfigureAwait(false);
+                await this.UpdateCollateralInfoAsync(this.cancellationSource.Token).ConfigureAwait(false);
 
-                if (!success)
-                {
-                    this.logger.LogWarning("Failed to update collateral. Ensure that mainnet gateway node is running and API is enabled. " +
-                                       "Node will not continue initialization before another gateway node responds.");
-
-                    try
-                    {
-                        await Task.Delay(CollateralInitializationUpdateIntervalSeconds * 1000, this.cancellationSource.Token).ConfigureAwait(false);
-                    }
-                    catch (OperationCanceledException)
-                    {
-                        this.logger.LogTrace("(-)[CANCELLED]");
-                        return;
-                    }
-                }
-                else
-                {
-                    break;
-                }
+                await this.DelayCollateralCheckAsync().ConfigureAwait(false);
             }
 
             this.updateCollateralContinuouslyTask = this.UpdateCollateralInfoContinuouslyAsync();
-            this.isInitialized = true;
         }
 
         /// <summary>Continuously updates info about money deposited to fed member's addresses.</summary>
         private async Task UpdateCollateralInfoContinuouslyAsync()
         {
-            while (!this.cancellationSource.Token.IsCancellationRequested)
+            while (!this.cancellationSource.IsCancellationRequested)
             {
-                try
-                {
-                    await Task.Delay(CollateralUpdateIntervalSeconds * 1000, this.cancellationSource.Token).ConfigureAwait(false);
-                }
-                catch (OperationCanceledException)
-                {
-                    this.logger.LogTrace("(-)[CANCELLED]");
-                    return;
-                }
+                await this.UpdateCollateralInfoAsync(this.cancellationSource.Token).ConfigureAwait(false);
+
+                this.logger.LogWarning("Node initialization will not continue until the gateway node responds.");
```

Comment:
here we are not initializing the node. Also warning will be displayed in case we managed to update collateral.

Reply:
Ah fail. Will fix.